[CORE] Fix missing build information

PHILO-HE opened the pull request:
## What changes were proposed in this pull request?

Fix issue reported here: https://github.com/apache/incubator-gluten/issues/10102#issuecomment-3095015236.
The build information related to backend is missing, such as GCC version,  backend branch/revision/revision time, etc. In the log and web UI, `"<unknown>"` is displayed for these items. 

The build information is added in two places during the compile phase: `gluten-core/pom.xml` and `gluten-substrait/pom.xml`. And they both modify `gluten-build-info.properties` placed in gluten-core's resource path: `gluten-core/target/generated-resources/`.

When building gluten-core module, the generated resources will be copied into target/classes/ and then be packed into module jar. In the later build for gluten-substrait module, the backend related build info is just added in the property file under gluten-core's resource path. And at the final package phase for the project, the shade plugin just packs classes and resource files inside module jars, not considering a module's resource path, which means the later added build info is not actually covered by the property file packed into Gluten fat jar.

In this pr, gluten-core's resource file is included in package module, and a configuration is added for shade plugin to exclude the property file comes from gluten-core jar.

## How was this patch tested?

Verified by local test.

A-little-bit-of-data commented:
@PHILO-HE   
Hello, I also compiled it under X86, and an error occurred at the last step.

Compile script :`./dev/builddeps-veloxbe.sh --run_setup_script=OFF --enable_s3=ON --enable_gcs=OFF --enable_abfs=OFF --enable_vcpkg=ON --build_arrow=OFF &&     mvn clean package -Pbackends-velox -Pspark-3.5 -DskipTests`



```
[647/647] Building CXX object velox/connectors/fuzzer/CMakeFiles/velox_fuzzer_connector.dir/FuzzerConnector.cpp.o
make[1]: Leaving directory '/incubator-gluten/ep/build-velox/build/velox_ep'
+ '[' -d _build/release/_deps ']'
+ echo 'Successfully built Velox from Source.'
Successfully built Velox from Source.
+ build_gluten_cpp
+ echo 'Start to build Gluten CPP'
Start to build Gluten CPP
+ cd /incubator-gluten/dev/../cpp
+ rm -rf build
+ mkdir build
+ cd build
+ GLUTEN_CMAKE_OPTIONS='-DBUILD_VELOX_BACKEND=ON     -DCMAKE_BUILD_TYPE=Release     -DVELOX_HOME=/incubator-gluten/dev/../ep/build-velox/build/velox_ep     -DBUILD_TESTS=OFF     -DBUILD_EXAMPLES=OFF     -DBUILD_BENCHMARKS=OFF     -DENABLE_JEMALLOC_STATS=OFF     -DENABLE_HBM=OFF     -DENABLE_QAT=OFF     -DENABLE_IAA=OFF     -DENABLE_GCS=OFF     -DENABLE_S3=ON     -DENABLE_HDFS=OFF     -DENABLE_ABFS=OFF'
+ '[' Linux == Darwin ']'
+ cmake -DBUILD_VELOX_BACKEND=ON -DCMAKE_BUILD_TYPE=Release -DVELOX_HOME=/incubator-gluten/dev/../ep/build-velox/build/velox_ep -DBUILD_TESTS=OFF -DBUILD_EXAMPLES=OFF -DBUILD_BENCHMARKS=OFF -DENABLE_JEMALLOC_STATS=OFF -DENABLE_HBM=OFF -DENABLE_QAT=OFF -DENABLE_IAA=OFF -DENABLE_GCS=OFF -DENABLE_S3=ON -DENABLE_HDFS=OFF -DENABLE_ABFS=OFF ..
-- Building using CMake version: 3.28.3
-- The C compiler identification is GNU 11.2.1
-- The CXX compiler identification is GNU 11.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/rh/devtoolset-11/root/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/rh/devtoolset-11/root/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Add definition NDEBUG
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Found JNI: /usr/lib/jvm/java-1.8.0-openjdk/include  found components: AWT JVM 
-- Found glog: /incubator-gluten/dev/vcpkg/vcpkg_installed/x64-linux-avx/lib/libglog.a  
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.27.1") 
Core module final CMAKE_CXX_FLAGS= -Wno-class-memaccess -Wno-error=unused-but-set-variable       -Wall        -Wno-sign-compare        -Wno-comment        -Werror        -Wno-error=parentheses        -Wno-error=unused-function        -Wno-error=unused-variable        -Wno-strict-aliasing        -Wno-ignored-qualifiers        -Wno-deprecated-declarations        -Wno-attributes -mavx2 -mfma -mavx -mf16c -mlzcnt -mbmi2
-- Found Boost: /incubator-gluten/dev/vcpkg/vcpkg_installed/x64-linux-avx/include (found version "1.84.0")  
-- Set Gluten Proto Directory in /incubator-gluten/gluten-core/src/main/resources/org/apache/gluten/proto
-- Set Substrait Proto Directory in /incubator-gluten/gluten-substrait/src/main/resources/substrait/proto
-- Found Arrow library: /usr/local/lib64/libarrow.a
CMake Error at CMake/ConfigArrow.cmake:44 (message):
  Arrow library Not Found: libparquet.a
Call Stack (most recent call first):
  core/CMakeLists.txt:180 (find_arrow_lib)
```

PHILO-HE commented:
@A-little-bit-of-data, if you are building Gluten early versions with latest docker image, please add `--build_arrow=ON` to re-install all required arrow libs, including libparquet.a.

In gluten main branch, we have removed the dependency on libparquet.a and also removed it from arrow build and installation. Then, latest docker image doesn't contain libparquet.a in its system lib path. Therefore, if older gluten version that still requires libparquet.a, you have to enable that arrow option.

A-little-bit-of-data commented:
> @A-little-bit-of-data, if you are building Gluten early versions with latest docker image, please add `--build_arrow=ON` to re-install all required arrow libs, including libparquet.a.
> 
> In gluten main branch, we have removed the dependency on libparquet.a and also removed it from arrow build and installation. Then, latest docker image doesn't contain libparquet.a in its system lib path. Therefore, if older gluten version that still requires libparquet.a, you have to enable that arrow option.

@PHILO-HE 
After recompiling and restarting, the log is as follows:
```
 ==============================================================

 Components: Velox

 Component Velox Branch: HEAD

 Component Velox Revision: eeb59ff4e1d298ad41e5dab54874fd03dee95481

 Component Velox Revision Time: 2025-05-12 17:18:16 +0800

 Gluten Version: 1.4.0

 GCC Version: GCC: (GNU) 11.2.1 20220127 (Red Hat 11.2.1-9)

 Java Version: <unknown>

 Scala Version: 2.12.15

 Spark Version: 3.5.2

 Hadoop Version: 3.3.4

 Gluten Branch: branch-1.4

 Gluten Revision: 50dd117dadb9fe3e9bba5c5002168e1294b2e2bb

 Gluten Revision Time: 2025-06-06 15:51:55 +0800

 Gluten Build Time: 2025-07-24T06:49:37Z

 Gluten Repo URL: https://ghfast.top/https://github.com/apache/incubator-gluten.git
```

PHILO-HE commented:
@A-little-bit-of-data, in my test with main branch, the java version is correctly displayed. I note you are using 1.4.0. It is because an extra space is appended for "--java" at the line below, which has already been fixed in main branch.

https://github.com/apache/incubator-gluten/blob/v1.4.0/gluten-core/pom.xml#L182
